compiler/rustc_hir_typeck/src/pat.rs

```diff
@@ -990,7 +990,7 @@ impl<'a, 'tcx> FnCtxt<'a, 'tcx> {
     fn check_pat_struct(
         &self,
         pat: &'tcx Pat<'tcx>,
-        qpath: &hir::QPath<'tcx>,
+        qpath: &'tcx hir::QPath<'tcx>,
```

Comment:
why did this change?

Reply:
```
error[E0621]: explicit lifetime required in the type of `qpath`
    --> compiler/rustc_hir_typeck/src/pat.rs:1000:39
     |
993  |         qpath: &hir::QPath<'tcx>,
     |                ----------------- help: add explicit lifetime `'tcx` to the type of `qpath`: `&'tcx QPath<'tcx>`
...
1000 |         let (variant, pat_ty) = match self.check_struct_path(qpath, pat.hir_id) {
     |                                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ lifetime `'tcx` required
```

Reply:
Yeah, and you added that requirement for `'tcx` on `check_struct_path` too. Why?

Reply:
Must have been due to some change that was no longer needed, got reverted and that I lost on the earlier gnarly rebase. Reverting.